synapse/groups/groups_server.py

```diff
@@ -598,7 +599,18 @@ def invite_to_group(self, group_id, user_id, requester_user_id, content):
         )
 
         # TODO: Check if user knocked
-        # TODO: Check if user is already invited
+
+        invited_users = yield self.store.get_invited_users_in_group(group_id)
+        if user_id in invited_users:
+            raise SynapseError(
+                403, "User already invited to group", errcode=Codes.BAD_STATE
```

Comment:
I think its a bad request rather than a lack of permissions.

```suggestion
                400, "User already invited to group", errcode=Codes.BAD_STATE
```